src/journal/journald-syslog.c

```diff
@@ -328,29 +333,37 @@ void server_process_syslog_message(
         struct iovec iovec[N_IOVEC_META_FIELDS + 6];
         unsigned n = 0;
         int priority = LOG_USER | LOG_INFO;
-        _cleanup_free_ char *identifier = NULL, *pid = NULL;
+        _cleanup_free_ char *identifier = NULL, *pid = NULL, *m = NULL;
+        char *syslog_message;
         const char *orig;
 
         assert(s);
         assert(buf);
 
         orig = buf;
+
         syslog_parse_priority(&buf, &priority, true);
 
         if (s->forward_to_syslog)
-                forward_syslog_raw(s, priority, orig, ucred, tv);
+                forward_syslog_raw(s, priority, orig, buf_len, ucred, tv);
+
+        syslog_message = m = memdup(buf, buf_len);
```

Comment:
hm, `memdup` calls `malloc` and `memcopy`. I think we shouldn't do it for **every** log message.
Why not https://github.com/systemd/systemd/pull/4856#issuecomment-267976998 ?